Automattic/jetpack
# wp-admin Site Default: Add a new site option for WPCOM admin interface

Related to https://github.com/Automattic/dotcom-forge/issues/4154
Related to: D125015-code

## Proposed changes:

This PR adds a new site option `wpcom_admin_interface` for WP.com sites to allow the user to set the admin interface on an Atomic site. 

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






**To test for Atomic site:** 

* Apply this diff (D125015-code) to your sandbox (if not deployed)
* Run `bin/jetpack-downloader test jetpack add/new-constant-for-wpcom-admin-interface` on your sandbox and apply it on top of the diff mentioned above
* Navigate to https://developer.wordpress.com/docs/api/console/
* Select `WP REST API` `wpcom/v2` `POST`
* Append `/sites/$site/hosting/admin-interface?interface=wp-admin` (replace `$site` with siteId)
* Make API request
* Navigate to https://developer.wordpress.com/docs/api/console/
* Select `WP.COM API`, `v1.1`, `GET` 
* Append `/sites/$site` (replace `$site` with the ID of the site that you used above)
* Make API request
* Confirm that the site options contain `wp_admin_interface` field:

<img width="1096" alt="Screenshot 2023-10-16 at 2 40 27 PM" src="https://github.com/Automattic/jetpack/assets/25575134/bc9bff16-3961-4a06-8cff-288471b58298">

**To test WoA site:** 

* Follow the https://github.com/Automattic/jetpack/pull/33610#issuecomment-1762886984
* Once the branch is activated, navigate around the dashboard and confirm that things generally work as expected and nothing seems to be broken

**To test on a Jetpack site:** 

Generally these changes should not touch self-hosted Jetpack sites but to be sure, feel free to run the steps below:
* Make sure that you have a Jetpack site connected to WP.com (you can use https://jurassic.ninja/ to create one)
* Follow the https://github.com/Automattic/jetpack/pull/33610#issuecomment-1762886984
* Once the branch is activated, navigate around the dashboard and confirm that things generally work as expected and nothing seems to be broken
* Try using the endpoint above to set the interface option for a Jetpack site (self-hosted) and confirm it is not possible
